SingularityUI/app/collections/TaskFiles.coffee

```diff
@@ -53,6 +53,24 @@ class TaskFiles extends Collection
 
         taskFiles
 
+    sortBy: (field, sortDirectionAscending) ->
+        if field is 'name'
+            sorted = _.sortBy @models, (file) =>
+                if file.attributes.isDirectory
```

Comment:
If we want to ignore case (because capital letters are always sorted before lowercase), do a .toLowercase() conversion in this function.